.github/workflows/ng-env-replace.js

```diff
@@ -0,0 +1,16 @@
+const fs = require('fs');
+
+const targetPath = './src/environments/environment.prod.ts';
+const file = fs.readFileSync(targetPath, 'utf8');
+const newFile = file.replace('GOOGLE_MAPS_API_KEY', process.env.GOOGLE_MAPS_API_KEY)
+    .replace('FIREBASE_CONFIG_API_KEY', process.env.FIREBASE_CONFIG_API_KEY)
+    .replace('FIREBASE_CONFIG_AUTH_DOMAIN', process.env.FIREBASE_CONFIG_AUTH_DOMAIN)
+    .replace('FIREBASE_CONFIG_DATABASE_URL', process.env.FIREBASE_CONFIG_DATABASE_URL)
+    .replace('FIREBASE_CONFIG_PROJECT_ID', process.env.FIREBASE_CONFIG_PROJECT_ID)
+    .replace('FIREBASE_CONFIG_STORAGE_BUCKET', process.env.FIREBASE_CONFIG_STORAGE_BUCKET)
+    .replace('FIREBASE_CONFIG_MESSAGING_SENDER_ID', process.env.FIREBASE_CONFIG_MESSAGING_SENDER_ID)
+    .replace('FIREBASE_CONFIG_APP_ID', process.env.FIREBASE_CONFIG_APP_ID)
+    .replace('CLOUD_FUNCTIONS_URL', process.env.CLOUD_FUNCTIONS_URL)
+    .replace('OFFLINE_BASE_MAP_SOURCES_URL', process.env.OFFLINE_BASE_MAP_SOURCES_URL);
+
+fs.writeFileSync(targetPath, newFile);
```

Comment:
Won't this update the actual source file, not the build?

Perhaps it would it be simpler to create a separate `environment.cd.ts` that uses refers directly to `process.env` as `environment.prod.ts` did, and we can let `environment.prod.ts` refer to the .keyfiles for local builds?